db/document.go

```diff
@@ -419,6 +420,10 @@ func UnmarshalDocumentSyncData(data []byte, needHistory bool) (*SyncData, error)
 		root.SyncData = &SyncData{History: make(RevTree)}
 	}
 	if err := base.JSONUnmarshal(data, &root); err != nil {
+		// note jsoniter validate will not validate number, string, boolean, array types
+		if json.Valid(data) {
```

Comment:
The overhead of json.Valid (another document scan) is relatively high for this function.  Under what situations will this be triggered?
1. Initial import of a standard bucket document (JSON object)?
2. Any attempt to import a non-JSON object?
3. Import of a standard bucket document with non-standard _sync property?

If it's triggered for either 1 or 2, I think the potential impact to import throughput is too high, as there are realistic deployments with high volumes of both of these.

Reply:
Also, it doesn't feel like this is the right place for us to be checking whether the document body is a JSON object.  (i.e. as part of the "backward compatibility check for legacy sync metadata in the document body").

Reply:
I think it's not necessary or helpful so I've removed this. This code would be triggered in import if there were no sync xattr so it actually would happen quite frequently. In this case we are looking for inline sync data to possibly migrate, so we do a partial unmarshal of the document to find if we should do that.

Independently, this is probably an area that could be optimized because this means for most SDK-only writes we are unmarshaling the document _twice_, once to find just the `SyncData` and another time when we decide to import it (or not) later.